go/marshal/encode.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Package marshal implements encoding and decoding of Noms values. The mapping between JSON objects and Go values is described  in the documentation for the Marshal and Unmarshal functions.
+package marshal
+
+import (
+	"reflect"
+	"runtime"
+	"sort"
+	"strings"
+	"sync"
+	"unicode"
+
+	"github.com/attic-labs/noms/go/types"
+)
+
+// Marshal converts a Go value to a Noms value.
+//
+// Marshal traverses the value v recursively. Marshal uses the following type-dependent encodings:
+//
+// Boolean values are encoded as Noms types.Bool.
+//
+// Floating point and integer values are encoded as Noms types.Number. At the moment this might lead to some loss in precision because types.Number currently takes a float64.
+//
+// String values are encoded as Noms types.String.
+//
+// Struct values are encoded as Noms structs (types.Struct). Each exported struct field becomes a member of the struct unless
```

Comment:
In this paragraph, it'd be good to make clear whether you mean a Noms Struct or a Go struct every time you use "struct"